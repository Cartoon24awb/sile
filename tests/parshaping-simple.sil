\begin[papersize=a7]{document}
\nofolios
\script[src=packages/lorem]
\begin{script}

local testshape = {
  { 60, 30, nil },
  { 40, 50, nil },
  { 0, 120, nil },
  { 10, 30, nil },
  { 10, nil, 90 },
  { 10, nil, 40 },
}

SILE.settings.set("linebreak.tolerance", 2000)
SILE.settings.set("document.baselineskip", "1.2em")
SILE.settings.set("document.parindent", 0)
SILE.settings.set("linebreak.parShape", true)

SILE.linebreak.parShape = function (self, line)
  local shape = testshape[line]
	if shape then
		return shape[1], shape[2], shape[3]
	else
		return nil, nil, nil
	end
end

SILE.call("lorem", { words = 30 })

\end{script}
\end{document}
