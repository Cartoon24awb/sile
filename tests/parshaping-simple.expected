Set paper size 	209.7637818	297.6377985
Begin page
Mx 	70.4882
My 	22.4502
Set font 	Gentium Plus;10;400;;normal;;LTR
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	96.7473
T	76 83 w=8.0078	(ip)
Mx 	104.7551
T	16 w=3.3691	(-)
Mx 	50.4882
My 	34.4502
T	86 88 80 w=17.1924	(sum)
Mx 	69.5891
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	93.4263
T	86 76 87 w=10.0146	(sit)
Mx 	10.4882
My 	46.4502
T	68 80 72 87 w=20.6836	(amet)
Mx 	33.0803
T	70 82 81 86 72 87 72 87 88 85 w=44.1602	(consetetur)
Mx 	79.1490
T	86 68 71 76 83 86 70 76 81 74 w=42.9980	(sadipscing)
Mx 	124.0556
T	72 79 76 87 85 w=17.4414	(elitr)
Mx 	20.4882
My 	58.4502
T	86 72 71 w=13.6816	(sed)
Mx 	36.0784
T	71 76 68 80 w=20.5322	(diam)
Mx 	20.4882
My 	70.4502
T	81 82 81 88 80 92 w=34.3262	(nonumy)
Mx 	57.5381
T	72 76 85 80 82 71 w=29.5508	(eirmod)
Mx 	89.8127
T	87 72 80 w=16.0938	(tem)
Mx 	105.9065
T	16 w=3.3691	(-)
Mx 	20.4882
My 	82.4502
T	83 82 85 w=14.2871	(por)
Mx 	38.7427
T	76 81 89 76 71 88 81 87 w=35.3271	(invidunt)
Mx 	78.0373
T	88 87 w=8.7402	(ut)
Mx 	90.7450
T	79 68 69 82 85 72 w=25.9863	(labore)
Mx 	120.6988
T	72 87 w=8.0615	(et)
Mx 	132.7277
T	71 82 79 82 85 72 w=26.5479	(dolore)
Mx 	10.4882
My 	94.4502
T	80 68 74 81 68 w=27.6074	(magna)
Mx 	42.7145
T	68 79 76 84 88 92 68 80 w=38.0127	(aliquyam)
Mx 	85.3460
T	72 85 68 87 w=16.6113	(erat)
Mx 	106.5762
T	86 72 71 w=13.6816	(sed)
Mx 	124.8767
T	71 76 68 80 w=20.5322	(diam)
Mx 	150.0278
T	89 82 79 88 83 87 88 68 w=36.5967	(voluptua)
Mx 	191.2434
T	68 87 w=8.0322	(at)
Mx 	10.4882
My 	106.4502
T	89 72 85 82 w=18.5400	(vero)
Mx 	31.6813
T	72 82 86 w=13.5107	(eos)
Mx 	47.8450
T	72 87 w=8.0615	(et)
Mx 	58.5596
T	68 70 70 88 86 68 80 w=35.1123	(accusam)
Mx 	96.3249
T	72 87 w=8.0615	(et)
End page
Finish
